rest/models/tickers.go

```diff
@@ -347,3 +349,46 @@ type TickerType struct {
 	Description string `json:"description,omitempty"`
 	Locale      string `json:"locale,omitempty"`
 }
+
+// GetTickerEventsParams is the set of parameters for the GetTickerEvents method.
+type GetTickerEventsParams struct {
+	// ID Identifier of an asset. This can currently be a Ticker, CUSIP, or Composite FIGI.
+	// When given a ticker, we return events for the entity currently represented by that ticker.
+	// To find events for entities previously associated with a ticker, find the relevant identifier
+	// using the Ticker Details Endpoint (https://polygon.io/docs/stocks/get_v3_reference_tickers__ticker).
+	ID string `validate:"required" path:"id"`
+
+	// A comma-separated list of the types of event to include. Currently, ticker_change is the only supported event_type. Leave blank to return all supported event_types.
+	Types *string `query:"types"`
+}
+
+func (p GetTickerEventsParams) WithTypes(types ...string) *GetTickerEventsParams {
+	q := strings.Join(types, ",")
+	p.Types = &q
+	return &p
+}
+
+// GetTickerEventsResponse is the response returned by the GetTickerEvents method.
+type GetTickerEventsResponse struct {
+	BaseResponse
+	Results []TickerEventResult `json:"results,omitempty"`
+}
+
+// TickerEventResult is the data for a ticker event.
+type TickerEventResult struct {
+	// Name is the company name.
+	Name   string        `json:"name"`
+	Events []TickerEvent `json:"events"`
+}
+
+// TickerEvent contains the data for the different type of ticker events that could occur.
+type TickerEvent struct {
+	Date         string             `json:"date"`
+	Type         string             `json:"type"`
+	TickerChange *TickerChangeEvent `json:"ticker_change,omitempty"`
```

Comment:
Does this need to be a pointer? It's fine if so but afaik it would be the only case where we use pointers in response types

Reply:
Oh interesting. Well for now I can leave it out. But there is a future where `ticker_change` isn't present in the response.